### DISCUSSION TITLE
Manage OVH IPFO blocks in NetBox with is_pool or Network/Broadcast restrictions

### DISCUSSION THREAD

**Discussion by dereckson** [netbox-community/netbox#16921] [Q&A] [answered]
The current model in NetBox for IPAM seems to be:

- if a IP prefix is a pool (is_pool = True), you can't assign an interface on any IP
- if an IP prefix is not a pool (is_pool = False), you can't assign an interface on the first and last IP of the range

For public IPv4 assignment, with IPv4 resources becoming more and more rare, IP blocks can be assigned **with the expectation to be used fully, without network and broadcast**. That's for example the case for IPFO (IP fail-over) from OVH to assign to VM on an hypervisor.

To add that block as prefix in NetBox has the advantage of being able to see what's still free in the block and when we need to reserve a new block, and also to pick a free IP address.

Previously, NetBox allowed to declare the block as a pool, and assign any IP to an interface. Now, it's not possible anymore, as documented in issues #12687 and #13392. Those issues are closed as "work as intended".

Now, we can't edit those first/last IPs anymore, regardless of the is_pool value.

How to best represent those IP blocks with current limitations?

**Previous issues**

- https://github.com/netbox-community/netbox/issues/9068 :: Validation of IP Addresses when of type network or broadcast #9068
- https://github.com/netbox-community/netbox/issues/12687 :: Change rules for 'Disallow assigning bcast/networks to interfaces'
- https://github.com/netbox-community/netbox/issues/13392 :: Unable to assign a Network/Broadcast IP to an Interface when Prefix is Pool

**Comment by markkuleinio** [comment]
How are those addresses configured in OVH, like 10.1.1.255/24 or 10.1.1.255/32?

**Reply by dereckson** [reply, marked as answer]
Prefix: 1.2.3.4/30, so we can see the free IPs in this.

IP: 1.2.3.4/30 too, which indeed doesn't make sense for the IPFO logic: the IP belongs to a range, but this range has no meaning as a subnetwork. Switching to 1.2.3.4/32 works.

Thanks :)
